### DISCUSSION TITLE
Simple Python Dependency won't work

### DISCUSSION THREAD

**Discussion by oosharma** [vercel/community#4673] [Help]
### Summary

I have an API route, it has a TS file which runs an ```exec``` function to run a python script which imports ```numpy```.

In my requirements.txt file, located at root level, I have added ```numpy==1.24.3```

When I push this code to GItHub, I get an error ```ImportError: No module named pyttsx3```

Github Link: https://github.com/oosharma/Next-App/tree/pip-freeze-attempt

### Example

Deployment Link: https://vercel.com/abhishek-sharmas-projects-03673e2d/easypost/DaAsC3QmPdautAHMQVPc3eFbyPqj

### Steps to Reproduce

Error message seen in deployment's logs: https://vercel.com/abhishek-sharmas-projects-03673e2d/easypost/DaAsC3QmPdautAHMQVPc3eFbyPqj

**Comment by misaelnieto** [comment]
Hi, The deployment link is not publicly visible. So this is only guesswork at this point.

Is your api route this one? https://next-app-lac-nine.vercel.app/api/runpyscript

If so, vercel runs your API route using the next.js runtime, which coincidentally has Python installed in it, but it was not designed nor prepared to run your python script with the requirements you are specifying on your requirements.txt. You would need a special runtime that has the python dependencies already installed, which might be a bit overkill to just run a small script. The option for you would be to run your python code as a serverless function. You can configure your serverless function to use the python runtime. This runtime will install the dependencies in your requirements.txt.

https://vercel.com/docs/functions/serverless-functions/runtimes/python

Look at the Django or Flask runtimes to get you started
